### PR TITLE
add support for generating basic-auth secret types

### PR DESCRIPTION
**Pull request by tomkennedy513**
- allow for optionally passing in username instead of generating

Co-authored-by: Aidan Obley <aobley@vmware.com>

### CODE REVIEW

File: pkg/kube/controllers/quarkssecret/quarkssecret_reconciler_test.go [hunk 1]
```diff
@@ -440,6 +441,110 @@ var _ = Describe("ReconcileQuarksSecret", func() {
...
+
+				result, err := reconciler.Reconcile(request)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(generator.GeneratePasswordCallCount()).To(Equal(2))
+				Expect(client.CreateCallCount()).To(Equal(1))
+				Expect(reconcile.Result{}).To(Equal(result))
+			})
+		})
+
+		When("username is provided", func() {
```
Comment (manno) [inline pkg/kube/controllers/quarkssecret/quarkssecret_reconciler_test.go:526]: Ah wow, and here I've been using `Context("when ...` all the time.
Thanks :+1:

File: integration/quarks_secret_test.go [hunk 2]
```diff
@@ -185,4 +185,28 @@ var _ = Describe("QuarksSecret", func() {
 			})
 		})
 	})
+
+	It("generates a basic auth secret with a username and password and deletes it when being deleted", func() {
+		// Create an QuarksSecret
```
Comment (manno) [inline integration/quarks_secret_test.go:190]: I also like using `By("creating a QuarksSecret")` instead of comments. Your choice.

**Comment by manno** [comment]
@tomkennedy513 sorry about the red integration test, we have this check if all examples have a test. Mostly works the other way around and checks if we remembered to increase the number...

File: integration/quarks_secret_test.go (hunk 2 above)
Comment (peterhaochen47) [reply integration/quarks_secret_test.go:190]: I think my teammates are trying to match the structure of the other test blocks in the same test file (`It` with comments). Perhaps we could leave this for now, and we could change all test blocks into `By` structure in the future if you prefer?

**Comment by peterhaochen47** [comment]
Added doc.

@manno Is there something we should do to fix the test failures? Is there a number we should bump somehwere?

**Comment by manno** [comment]
> @manno Is there something we should do to fix the test failures? Is there a number we should bump somewhere?

Hey, yes the tests fail here: e2e/kube/examples_count_test.go:12

**Pull request merged**